public/react/Theme/Theme.tsx

```diff
@@ -0,0 +1,100 @@
+import type { FC } from "react";
+
+import React, { useEffect, useState } from "react";
+import SidebarButton from "../components/SidebarButton";
+
+import ArrowLeftCircleIcon from "@atlaskit/icon/glyph/arrow-left-circle";
+import Heading from "@atlaskit/heading";
+import Lozenge from "@atlaskit/lozenge";
+import { getTextColorUsingAPCA } from "../../utils/color";
+import { useTheme } from "../services/theme/useTheme";
+import { useSaveTheme } from "../services/theme/useSaveTheme";
+import { applyThemeToCssVars, defaultTheme, Theme } from "../../jira/theme";
+import { useDebounce } from "../hooks/useDebounce";
+import Spinner from "@atlaskit/spinner";
+import Button from "@atlaskit/button/new";
+
+interface ThemeProps {
+  onBackButtonClicked: () => void;
+}
+
+const Theme: FC<ThemeProps> = ({ onBackButtonClicked }) => {
+  const theme = useTheme();
+  const { save, isPending } = useSaveTheme();
+
+  const [localTheme, setLocalTheme] = useState(theme);
+
+  const updateLocalTheme = (newLocalTheme: Theme) => {
+    applyThemeToCssVars(newLocalTheme);
+    setLocalTheme(newLocalTheme);
+  };
+
+  const debouncedTheme = useDebounce(localTheme, 500);
+
+  useEffect(() => {
+    if (JSON.stringify(theme) === JSON.stringify(debouncedTheme)) {
+      return;
+    }
+
+    save(debouncedTheme);
+  }, [debouncedTheme, save]);
+
+  return (
+    <>
+      <SidebarButton onClick={onBackButtonClicked}>
+        <ArrowLeftCircleIcon label="go back" />
+        Go back
+      </SidebarButton>
+      <div className="my-4">
+        <div className="flex justify-between">
+          <Heading size="small">Theme</Heading>
+          {isPending && <Spinner size="small" />}
+        </div>
+        <div className="pt-6 flex flex-col gap-8">
+          {Object.entries(localTheme).map(
```

Comment:
how are you preserving the order with an object?  Should the theme be an array so we make sure order is always preserved?

(Eventually, when there's a rules engine, it will certainly be in some sort of an array)

Reply:
it should be. I was going to bring it up tomorrow. Currently this implementation doesn't care about order. If it does, do we have a way to sort? 

We could do an array in app data but at some point we'd have to sort that too (eventually when there's a rules engine)